docs/Secure_Gateways_with_Istio

```diff
@@ -0,0 +1,124 @@
+# Secure Gateways with Istio
+
+## Prerequisite
+
+### Enable Istio
+
+Before installing a cluster yaml file with kubefate, please make sure you have enabled istio.
+
+```yaml
+istio:
+  enabled: false
+```
+
+
+
+### Prepare root certificate
+
+Create a root certificate and private key to sign the certificates for your services.
+
+`$ openssl req -x509 -sha256 -nodes -days 365 -newkey rsa:2048 -subj '/O=example Inc./CN=example.com' -keyout example.com.key -out example.com.crt`
+
+Now we get `example.com.key` and `example.com.crt` as CA to sign the certificates.
+
+### Set ingress IP and port
+
+If your environment has an external load balancer, set the ingress IP and ports:
+
+```bash
+$ export INGRESS_HOST=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+$ export INGRESS_PORT=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="http2")].port}')
+$ export SECURE_INGRESS_PORT=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="https")].port}')
+```
+
+If you're using node port, run:
+
+```bash
+$ export INGRESS_HOST=$(minikube ip)
+$ export INGRESS_PORT=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="http2")].nodePort}')
+$ export SECURE_INGRESS_PORT=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="https")].nodePort}')
+```
+
+Please NOTE: enable istio will disable nginx-ingress for fateboard and notebook, so you have to access them with $INGRESS_HOST and $SECURE_INGRESS_PORT.
+
+## Generate credential for your services
+
+1. Create a certificate and a private key for `*.fateboard.example.com`:
+
+   ```bash
+   $ openssl req -out fateboard.example.com.csr -newkey rsa:2048 -nodes -keyout fateboard.example.com.key -subj "/CN=*.fateboard.example.com/O=example organization"
```

Comment:
Can we do not use fateboard.example.com.csr as the name, that is too long, use a directory to do that.

Please check the doc about how to create TLS for eggroll:
https://github.com/FederatedAI/KubeFATE/blob/develop-1.9.0/docs/Eggroll_with_TLS.md